### PR TITLE
Initial changes for docker distribution

### PR DESCRIPTION
**Pull request by yadavcbala**
*Issue #, if available:*

*Description of changes:*
* Includes changes for basic Dockerfile
* Adds build file which runs pre-reqs and builds the docker image

#### Testing
*Below is the log of tasks run on `./gradlew clean :release:docker` shows the prerequisites are executed before preparing the docker image*
```
byadav$:simple-ingest-transformation-utility-pipeline byadav$ ./gradlew clean :release:docker --console=plain
=======================================
Elasticsearch Build Hamster says Hello!
  Gradle Version        : 6.5
  JDK Version           : 14 (Oracle JDK)
  JAVA_HOME             : /Library/Java/JavaVirtualMachines/jdk-14.0.2.jdk/Contents/Home
  Random Testing Seed   : D7AC0028FE3D005C
  In FIPS 140 mode      : false
=======================================
> Task :release:clean
> Task :research:clean UP-TO-DATE
> Task :situp-api:clean
> Task :situp-benchmarks:clean UP-TO-DATE
> Task :situp-core:clean
> Task :situp-plugins:clean
> Task :research:zipkin-elastic-to-otel:clean UP-TO-DATE
> Task :situp-benchmarks:lmdb-benchmarks:clean UP-TO-DATE
> Task :situp-benchmarks:mapdb-benchmarks:clean UP-TO-DATE
> Task :situp-benchmarks:service-map-stateful-benchmarks:clean UP-TO-DATE
> Task :situp-plugins:apmtracesource:clean
> Task :situp-plugins:common:clean
> Task :situp-plugins:elasticsearch:clean
> Task :situp-plugins:lmdb-processor-state:clean
> Task :situp-plugins:mapdb-processor-state:clean
> Task :situp-plugins:service-map-stateful:clean
> Task :release:dockerClean UP-TO-DATE
> Task :release:benchmarkTests UP-TO-DATE
> Task :situp-api:compileJava
> Task :situp-api:processResources NO-SOURCE
> Task :situp-api:classes
> Task :situp-api:jar
> Task :situp-plugins:common:compileJava
> Task :situp-plugins:common:processResources NO-SOURCE
> Task :situp-plugins:common:classes
> Task :situp-plugins:common:jar
> Task :situp-plugins:apmtracesource:compileJava
> Task :situp-plugins:apmtracesource:processResources NO-SOURCE
> Task :situp-plugins:apmtracesource:classes
> Task :situp-plugins:apmtracesource:jar
> Task :situp-plugins:elasticsearch:compileJava
> Task :situp-plugins:elasticsearch:processResources
> Task :situp-plugins:elasticsearch:classes
> Task :situp-plugins:elasticsearch:jar
> Task :situp-plugins:lmdb-processor-state:compileJava
> Task :situp-plugins:lmdb-processor-state:processResources NO-SOURCE
> Task :situp-plugins:lmdb-processor-state:classes
> Task :situp-plugins:lmdb-processor-state:jar

> Task :situp-plugins:mapdb-processor-state:compileJava
Note: /workplace/byadav/ForkedSITUP/simple-ingest-transformation-utility-pipeline/situp-plugins/mapdb-processor-state/src/main/java/com/amazon/situp/plugins/processor/state/MapDbProcessorState.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :situp-plugins:mapdb-processor-state:processResources NO-SOURCE
> Task :situp-plugins:mapdb-processor-state:classes
> Task :situp-plugins:mapdb-processor-state:jar
> Task :situp-plugins:service-map-stateful:compileJava
> Task :situp-plugins:service-map-stateful:processResources NO-SOURCE
> Task :situp-plugins:service-map-stateful:classes
> Task :situp-plugins:service-map-stateful:jar
> Task :situp-plugins:compileJava NO-SOURCE
> Task :situp-plugins:processResources NO-SOURCE
> Task :situp-plugins:classes UP-TO-DATE
> Task :situp-plugins:jar
> Task :situp-core:compileJava
> Task :situp-core:processResources
> Task :situp-core:classes
> Task :situp-core:jar
> Task :situp-core:assemble
> Task :situp-core:compileTestJava
> Task :situp-core:processTestResources
> Task :situp-core:testClasses
> Task :situp-core:test
> Task :situp-core:check
> Task :situp-core:jacocoTestReport
> Task :situp-core:build
> Task :release:buildCore
> Task :release:endToEndTests UP-TO-DATE
> Task :release:dockerPrerequisites
> Task :release:dockerPrepare
> Task :release:docker

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.5/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 1m 10s
42 actionable tasks: 35 executed, 7 up-to-date

```

Execute without JAVA_OPTS

```
byadav$ docker run --name situp3 --expose 21890 --read-only -v /workplace/byadav/multipipelines2.yml:/usr/share/situp/situp.yml situp/situp:0.1-beta
0    [main] INFO  com.amazon.situp.SitupExecute  – ======================================
10   [main] INFO  com.amazon.situp.SitupExecute  – Free memory:  111 MB
10   [main] INFO  com.amazon.situp.SitupExecute  – Allocated memory: 128 MB
11   [main] INFO  com.amazon.situp.SitupExecute  – Max memory: 1,994 MB
11   [main] INFO  com.amazon.situp.SitupExecute  – Total free memory: 1,977 MB
11   [main] INFO  com.amazon.situp.SitupExecute  – ======================================
12   [main] INFO  com.amazon.situp.Situp  – Using /usr/share/situp/situp.yml configuration file
```

Execute with JAVA_OPTS

```
byadav$ docker run --name situp3 -e JAVA_OPTS='-Xmx3g -Xms3g' --expose 21890 --read-only -v /workplace/byadav/multipipelines2.yml:/usr/share/situp/situp.yml situp/situp:0.1-beta
0    [main] INFO  com.amazon.situp.SitupExecute  – ======================================
6    [main] INFO  com.amazon.situp.SitupExecute  – Free memory:  3,055 MB
6    [main] INFO  com.amazon.situp.SitupExecute  – Allocated memory: 3,072 MB
7    [main] INFO  com.amazon.situp.SitupExecute  – Max memory: 3,072 MB
7    [main] INFO  com.amazon.situp.SitupExecute  – Total free memory: 3,055 MB
7    [main] INFO  com.amazon.situp.SitupExecute  – ======================================
8    [main] INFO  com.amazon.situp.Situp  – Using /usr/share/situp/situp.yml configuration file
```

The subsequent changes can be creating image file with clean OS and install required packages and use tarball.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: release/build.gradle [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+task buildCore {
+    dependsOn ':situp-core:build'
+}
+
+task dockerPrerequisites {
+    dependsOn 'buildCore'
+    dependsOn 'endToEndTests'
+    dependsOn 'benchmarkTests'
```
Comment (chenqi0805) [inline release/build.gradle:24]: QUES: Should benchmarkTests be a dependent task for docker build? I think benchmarkTests only measures performance but not fuctionality of SITUP.

File: release/build.gradle [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+docker {
+    name "odfe-pipes/${project.rootProject.name}:${project.version}"
+    tag "${project.name}", "${project.version}" //update tag when ready to push
```
Comment (chenqi0805) [inline release/build.gradle:29]: QUES: are the tags going to be updated inline or in a separate config file?

File: release/build.gradle [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    files project(':situp-core').jar.archivePath
+    buildArgs(['JAR_FILE': project(':situp-core').jar.archiveName,
+               'CONFIG_FILEPATH': '/usr/share/situp/odfe-pipes.yml'])
```
Comment (chenqi0805) [inline release/build.gradle:32]: QUES: is it expected user put config file in this exact location? Shall we allow custom file path?

File: release/build.gradle (hunk 1 above)
Comment (yadavcbala) [reply release/build.gradle:24]: Good point - for now it looked like we needed it as a prereq for release but as you see its an empty task to show that we can add pre-reqs of that sort if required.

File: release/build.gradle (hunk 2 above)
Comment (yadavcbala) [reply release/build.gradle:29]: after we decide the tag name pattern, we can update it either ways - would prefer to read it from a global file which has details about the release and appropriate release notes

File: release/build.gradle (hunk 3 above)
Comment (yadavcbala) [reply release/build.gradle:32]: Good Ques - it is the default location - user should be able to pass in the different location if he intends to - I will add that as part of next PR.

**Review by chenqi0805 — APPROVED** [review]
LGTM.

File: release/docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+FROM adoptopenjdk/openjdk14:jre-14.0.1_7-alpine
+ARG CONFIG_FILEPATH
+ARG JAR_FILE
+ENV ENV_CONFIG_FILEPATH=$CONFIG_FILEPATH
+ENV SITUP_PATH /usr/share/situp
+RUN mkdir -p $SITUP_PATH
+COPY $JAR_FILE /usr/share/situp/odfe-pipes.jar
+WORKDIR $SITUP_PATH
+CMD java -jar odfe-pipes.jar ${ENV_CONFIG_FILEPATH}
```
Comment (kowshikn) [inline release/docker/Dockerfile:12]: JVM args?

File: release/docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# building on CentOS 7 to be in line with
+# https://github.com/opendistro-for-elasticsearch/opendistro-build/blob/master/elasticsearch
+
+FROM adoptopenjdk/openjdk14:jre-14.0.1_7-alpine
+ARG CONFIG_FILEPATH
+ARG JAR_FILE
+ENV ENV_CONFIG_FILEPATH=$CONFIG_FILEPATH
+ENV SITUP_PATH /usr/share/situp
+RUN mkdir -p $SITUP_PATH
+COPY $JAR_FILE /usr/share/situp/odfe-pipes.jar
```
Comment (kowshikn) [inline release/docker/Dockerfile:10]: nit: lets stick to situp.jar, odfe-pipes is not the name we want.

File: release/docker/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# building on CentOS 7 to be in line with
+# https://github.com/opendistro-for-elasticsearch/opendistro-build/blob/master/elasticsearch
+
+FROM adoptopenjdk/openjdk14:jre-14.0.1_7-alpine
```
Comment (kowshikn) [inline release/docker/Dockerfile:4]: Java version?

File: release/build.gradle (hunk 3 above)
Comment (kowshikn) [reply release/build.gradle:32]: Users are expected to use the docker image, they will not building a new image. Why would users want custom path?

**Comment by kowshikn** [comment]
nit: avoid odfe-pipes usage.

File: release/build.gradle (hunk 3 above)
Comment (chenqi0805) [reply release/build.gradle:32]: @kowshikn Maybe I misunderstood this line. So is `CONFIG_FILEPATH` just a **default** location looking for the config file when user runs the image?

File: release/docker/Dockerfile (hunk 6 above)
Comment (yadavcbala) [reply release/docker/Dockerfile:4]: pending for next PR

**Review by kowshikn — APPROVED** [review]

**Pull request merged**